src/Tasks/Microsoft.Common.CurrentVersion.targets

```diff
@@ -3973,6 +3973,7 @@ Copyright (C) Microsoft Corporation. All rights reserved.
       <_DeploymentReferencePaths Include="@(_DeploymentReferencePaths);@(_CopyLocalFalseRefPathsWithExclusion)" />
       <_ManifestManagedReferences Include="@(_DeploymentReferencePaths);@(ReferenceDependencyPaths);@(_SGenDllsRelatedToCurrentDll);@(SerializationAssembly)"
                                Exclude="@(_SatelliteAssemblies);@(_ReferenceScatterPaths);@(_ExcludedAssembliesFromManifestGeneration)" />
+      <_ManifestManagedReferences Remove="@(_ManifestManagedReferences)" Condition="'%(CopyLocal)' == 'false'" />
```

Comment:
Did you check the scenario that a reference assembly has CopyLocal equal to false but it is manually included in the application files?

Reply:
> Do all the items in _ManifestManagedReferences item group have CopyLocal property?

All of the items should have the property as far as I can tell. I changed the syntax just in case to make sure the build won't break if an item doesn't have the property. Now only items that explicitly have the property set and the property value is false get removed.

I tested this locally by manually adding items to the collection that don't have the property set and observing the behavior.

Reply:
> Did you check the scenario that a reference assembly has CopyLocal equal to false but it is manually included in the application files?

I changed the code so we filter out dependencies after executing the ResolveManifestFiles task, so we have information about what settings are applied in the Application Files view per file.

Instead of filtering out all files that are not set CopyLocal, I now filter out files that have both CopyLocal set to False and DependencyType set to a value different than 'Install'.

That should cover the case where CopyLocal is set to false but in the Application Files window the user has manually set the file to Include.